src/approaches/grammar_search_invention_approach.py

```diff
@@ -514,28 +609,69 @@ def _get_successors(s: FrozenSet[Predicate]
 
 
 def _count_positives_for_ops(strips_ops: List[STRIPSOperator],
-                             segments: List[Segment]
-                             ) -> Tuple[int, int]:
-    """Returns num true positives, num false positives.
+                             option_specs: List[OptionSpec],
+                             segments: List[Segment],
+                             ) -> Tuple[int, int,
+                                        List[Set[int]], List[Set[int]]]:
+    """Returns num true positives, num false positives, and for each strips op,
+    lists of segment indices that contribute true or false positives.
     """
+    assert len(strips_ops) == len(option_specs)
     num_true_positives = 0
     num_false_positives = 0
-    for segment in segments:
+    # The following two lists are just useful for debugging with
+    # _run_debug_analysis.
+    true_positive_idxs : List[Set[int]] = [set() for _ in strips_ops]
+    false_positive_idxs : List[Set[int]] = [set() for _ in strips_ops]
+    for idx, segment in enumerate(segments):
         objects = set(segment.states[0])
-        ground_ops = [o for op in strips_ops
-                      for o in utils.all_ground_operators(op, objects)]
+        segment_option = segment.get_option()
+        option_objects = segment_option.objects
         covered_by_some_op = False
-        for ground_op in ground_ops:
-            if not ground_op.preconditions.issubset(segment.init_atoms):
+        # Ground only the operators with a matching option spec.
+        for op_idx, (op, option_spec) in enumerate(zip(strips_ops,
+                                                       option_specs)):
+            # If the parameterized options are different, not relevant.
+            if option_spec[0] != segment_option.parent:
                 continue
-            if ground_op.add_effects == segment.add_effects and \
-               ground_op.delete_effects == segment.delete_effects:
-                covered_by_some_op = True
-            else:
-                num_false_positives += 1
+            option_vars = option_spec[1]
+            assert len(option_vars) == len(option_objects)
+            option_var_to_obj = dict(zip(option_vars, option_objects))
+            # We want to get all ground operators whose corresponding
+            # substitution is consistent with the option vars for this
+            # segment. So, determine all of the operator variables
+            # that are not in the option vars, and consider all
+            # groundings of them.
+            types = [p.type for p in op.parameters
+                     if p not in option_vars]
+            for choice in utils.get_object_combinations(objects, types):
```

Comment:
Would this code be simpler if we did the same thing as before, but then filter out when the option arguments don't match?
```
for op in strips_ops:
    for ground_op in utils.all_ground_operators(op, objects):
        if the option_vars of the ground_op don't match the option_objects:
            continue
```

Just a thought... if we don't do this, then we should move the functionality you've got here into utils.py, maybe into a method called `all_ground_operators_consistent_with_option`

Reply:
I don't think we want to do the first thing because that would increase the computational complexity. I'll make the utility